_posts/2018-3-11-interactive-examples.md

```diff
@@ -0,0 +1,100 @@
+
+![](../images/interactive-examples/js-editor-final-user-test.png)
+
+![](../images/interactive-examples/css-editor-final-user-test.png)
+
+Then we had another round of user testing. This time we ran remote user tests over video, with participants recruited through MDN itself. This enabled us to have a tight feedback loop for the editors: we could quickly make and test adjustments based on user feedback.
+
+This time user testing was very positive, and we decided we were ready for beta testing.
+
+## Beta testing
+
+The beta test started at the end of August and lasted for two weeks. We embedded editors on three JS and three CSS pages, added a survey, and asked for feedback. [Danielle Vincent] mentioned it in the [MDN newsletter], which drove thousands of people to [our Discourse post announcing it].
+
+Feedback was overwhelmingly positive: 156/159 people who took the survey voted to see the editor on more pages, and the free-form text feedback was very encouraging. We were confident that we had a good UX.
+
+## JavaScript examples and page load optimization 
+
+Now we had an editor but very few actual examples. We asked [Mark Boas] to write examples for the JS reference pages, and in a couple of months he had written about 400 beautiful concise examples.
+
+<iframe width="100%" height="490px" frameborder="0" src="https://interactive-examples.mdn.mozilla.net/pages/js/array-slice.html"></iframe>
+
+We had another problem, though: it was clear that the editors regressed page load time too much. Schalk and Stephanie wrung every last millisecond of performance optimization out of the architecture, and finally, in December 2017, we decided to ship.
+
+In fact we're still not happy with the performance, but have some extra tricks planned for this year that we hope will make it back and more.
+
+## CSS examples
+
+In the first three weeks of 2018 Schalk and I updated 400 JavaScript pages to include Mark's examples, and then we turned to getting examples written for the CSS pages.
+
+[We asked for help], [Jen Simmons tweeted about it], and three weeks later [our community had contributed more than 150 examples], with over a hundred coming from a single volunteer, [mfluehr].
+
+<iframe width="100%" height="390px" frameborder="0" src="https://interactive-examples.mdn.mozilla.net/pages/css/rotate3d.html"></iframe>
+
+After that [Rachel Andrew] and [Daniel Beck] started working with us, and they took care of the rest.
+
+<iframe width="100%" height="390px" frameborder="0" src="https://interactive-examples.mdn.mozilla.net/pages/css/clip-path.html"></iframe>
+
+## What's next?
+
+Right now we're working on implementing interactive examples for the HTML reference. We just concluded a round of user testing, with [encouraging results], and hope to start writing examples soon.
+
+After that we'll look at the Web APIs. This is a huge area, so we'll prioritise some sections of it to start with.
```

Comment:
Do you want input on this prioritization? Seems like a natural opening to ask...

Reply:
Not really, we more or less know. We'll probably start off with the basic DOM APIs (e.g. Window, Document). Perhaps it would be more helpful to say that though.